### PR TITLE
feat: Add ClientOptions.SendDefaultPii

### LINKED ISSUE #175: PII opt-in by default
We need to add to the SDK a new config bool such that anything that we know will identify a user is not sent on events by default. The entire user object essentially, auth headers, IP addresses, that sort of thing.

See `send_default_pii` in the Python SDK.
https://docs.sentry.io/error-reporting/configuration/?platform=python#send-default-pii

The current docs for Go suggests that it exists as `SendDefaultPii`, but it doesn't exist yet.
https://docs.sentry.io/error-reporting/configuration/?platform=go#send-default-pii

(see also https://github.com/getsentry/develop/issues/466)

### PR DESCRIPTION
**Pull request by cleptric**
As `scope.SetUser` has to be explicitly called by the user, I did not add a check for `SendDefaultPii` there. 

Closes #175

### CODE REVIEW

**Comment by cleptric** [comment]
- I've noticed that the PHP SDK still sends all headers but sets their value to `[Filtered]`.
Other SDKs drop the keys completely.
- Do I need to add checks for the request body as well?

**Comment by AbhiPrasad** [comment]
> Do I need to add checks for the request body as well?

Yes, it's a source of PII as well

**Comment by cleptric** [comment]
While the `Request` struct contains a `Data` field, I can't find any usage of this field in the SDK. So we're good to go with this PR.

**Review by AbhiPrasad — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,11 @@
 # Changelog
 
+## Unreleased
+
 - fix: Scope values should not override Event values (#446)
 - feat: Extend User inteface by adding Data, Name and Segment (#483)
 - feat: Make maximum amount of spans configurable (#460)
+- feat: Add ClientOptions.SendDefaultPii #485
```
Comment (rhcarvalho) [inline CHANGELOG.md:8]: Happy to see the Sentry Go SDK evolving!

One suggestion: initialisms like "PII" are spelled with a consistent case as per Go community conventions https://github.com/golang/go/wiki/CodeReviewComments#initialisms.

The option could be called `ClientOptions.SendDefaultPII`.

Comment (cleptric) [reply CHANGELOG.md:8]: Thanks, I opened https://github.com/getsentry/sentry-go/pull/490

**Pull request merged**